fw/http_frame.c

```diff
@@ -1951,7 +1994,37 @@ tfw_h2_frame_process(TfwConn *c, struct sk_buff *skb, struct sk_buff **next)
 			kfree_skb(nskb);
 			goto out;
 		}
-	} else {
+	}
+	else if (unlikely(tfw_h2_allowed_empty_frame(h2))) {
```

Comment:
I have a question here, i am not sure. Can we use previous path of code for our purpose? and pass NULL skb in tfw_http_msg_process_generic, when we deal with empty data/header/continuation frame?

Reply:
Or you can alloc skb with zero data ss_skb_alloc(0) in case when if (!h2->skb_head)  and tfw_h2_allowed_empty_frame